static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -1,200 +1,253 @@
+};
+
+const yScaleFunc = (padding, height, max) => {
+  return scaleLinear()
+    .rangeRound([height - padding.top - padding.bottom, padding.top]).nice()
+    .domain([0, max + Math.ceil(max * 0.1)]);
+};
+
+const colourScaleFunc = (keys) => {
+  return scaleOrdinal(schemePaired)
+    .domain(keys);
+};
+
+const getStackedData = (keys, data) => {
+  const stackFunc = stack()
+    .order(stackOrderReverse)
+    .keys(keys);
+
+  return stackFunc(data);
+};
+
+const bisectDate = bisector(d => d.date).left;
+
+const tooltipRows = (data, currentHoverKey, colorScale) => {
+  let dataArr = [];
+  let total = 0;
+  let other = {
+    key: 'other',
+    value: 0,
+    count: 0
+  };
+
+  Object.keys(data)
+    .filter(key => key !== 'date')
+    .forEach(key => {
+      total += data[key];
+
+      dataArr.push({
+        key: key,
+        value: data[key]
+      });
+    });
 
-    while(offset < 30) {
-      days.push(start.clone().add(offset, 'days'));
-      dummyActiveDevices.push(0);
-      offset++;
-    }
-    activeDevices.push(dummyActiveDevices);
-    minYAxisValue = 1;
+  dataArr.sort((a, b) => b.value - a.value);
+
+  // Filter out anything below 0.1% of the total users
+  // If there are more than 10 series
+  if (dataArr.length > 10) {
+    dataArr.forEach(item => {
+      if (item.value / total < 0.001) {
+        other.value += item.value;
+        other.count += 1;
+        other.key = `${other.count} other`;
+        item.skip = true;
+      }
+    });
   }
 
-  activeDevices.sort((a, b) => {
-    const a1 = a.slice(1); // Remove the first item as it's the label
-    const b1 = b.slice(1); // Same again.
+  // If we've added anything to the 'other' series,
+  // add it to the array
+  if (other.value > 0) {
+    dataArr.push(other);
+  }
 
-    const reducer = (accumulator, currentValue) => {
-      return accumulator + currentValue;
+  return dataArr.map(item => {
+    if (!item.skip) {
+      return `<span class="snapcraft-graph-tooltip__series${item.key === currentHoverKey ? ' is-hovered' : ''}" title="${item.key}">
+          <span class="snapcraft-graph-tooltip__series-name">${item.key}</span>
+          <span class="snapcraft-graph-tooltip__series-color"${!item.count ? `style="background:${colorScale(item.key)};"` : ''}></span>
+          <span class="snapcraft-graph-tooltip__series-value">${commaValue(item.value)}</span>
+        </span>`;
+    }
+  }).join('');
+};
+
+const tooltipTemplate = (data, currentHoverKey, colorScale) => {
+  return singleLine(`<div class="p-tooltip p-tooltip--top-center" style="display: block;">\
+      <span class="p-tooltip__message" role="tooltip" style="display: block;">\
+        <span class="snapcraft-graph-tooltip__title">${tooltipTimeFormat(data.date)}</span>\
+        ${tooltipRows(data, currentHoverKey, colorScale)}\
+      </span>\
+    </div>`);
+};
+
+function drawGraph(activeDevices) {
+  // Basic svg setup
+  const holder = document.getElementById('activeDevices');
```

Comment:
While we are changing this it would be good not to hardcode ids into JS, just pass them to the function like we do in many other places.

Reply:
Good point, will fix